### PR TITLE
Refactor codebase to save/load data from JSON files on app startup/exit

### PR DESCRIPTION
**Pull request by didierganthier**
### Summary
This PR refactors the App class to save and load data from JSON files in a more efficient and readable way. The code is now shorter and easier to maintain.

### Checklist

- [x]  Refactor save_data method to separate saving of each data type into their own methods
- [x]  Refactor load_data method to separate loading of each data type into their own methods
- [x]  Make sure all data is still saved and loaded correctly

### What changes have been made?
- save_data method now calls separate methods save_persons, save_books, and save_rentals to save each data type to their respective JSON files
- load_data method now calls separate methods read_persons, read_books, and read_rentals to load each data type from their respective JSON files
- Added helper methods save_persons, save_books, save_rentals, read_persons, read_books, read_rentals to handle saving and loading of each data type
- Removed unnecessary code and made use of File.open with block to automatically close file after reading or writing to it

Authors:
- @didierganthier 
- @cvalencia1991

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -15,6 +16,78 @@
...
+  def read_rentals
+    return [] unless File.exist?('rentals.json')
+
+    rentals_json = JSON.parse(File.read('rentals.json'))
+    rentals_json.map do |rental|
+      Rental.new(rental['date'], @person[rental['person_index']], @books[rental['book_index']])
+    end
+  end
+
   def print_question
```
Comment (sja-thedude) [inline app.rb:91]: - [ ] Good job so far, however when I enter person, book or rental it is not added to the JSON files. The requirement is that the data added should be saved in the respective files and preserved even when you exit the app. Kindly fix it.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @didierganthier & @cvalencia1991, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Correct Gitflow
✔️ Professional Readme

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (cvalencia1991) [reply app.rb:91]: could you please do it again, it save it  @sja-thedude

Comment (cvalencia1991) [reply app.rb:91]: sorry @sja-thedude  but the requirements say's `On exit from your app save all the data (books, people, and rentals) into JSON files.` no, it adds admittedly, I think you qualify wrong.

**Comment by didierganthier** [comment]
For anyone else who happens to be reviewing this, I believe the first reviewer made a mistake. I tried to reach out too her but she wouldn't listen. I even have a Loom Video to prove that it's working. Thanks. :)

https://loom.com/share/0c05d80e116c441fad0b318692e480bc

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @didierganthier and @cvalencia1991,

✔️ Implementation of data storage on app exit

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by didierganthier** [comment]
Thanks @TedLivist 👍🏽

**Pull request merged**